src/metrics/peerstore.go

```diff
@@ -359,6 +361,41 @@ func (c *PeerStore) ExportToCSV(filePath string) error {
 	return nil
 }
 
+// Function that save the raw Data of all the peers to a db (Using BoltDB)
+// Using the peerID as a key, and the Peer struct marshalled into JSON as a value
+func (c *PeerStore) PersistToDb() {
+	db, err := bolt.Open("../../peerstore.db", 0600, &bolt.Options{Timeout: 5 * time.Second})
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+
+	err = db.Update(func(tx *bolt.Tx) error {
+		b, err := tx.CreateBucketIfNotExists([]byte("PeerStore"))
+		if err != nil {
+			return err
+		}
+
+		c.PeerStore.Range(func(k, val interface{}) bool {
+			peerData := val.(Peer)
+			buf, err := json.Marshal(peerData)
```

Comment:
How does the marshalled information of a given peer look like? Is it a readable json? I thought we needed to write some annotations in order to work, something like [this](https://gist.github.com/AlexMocioi/10008253)

Reply:
The JSON look like this
```
{
   "PeerId":"16Uiu2HAkvPRqnUZx2zKwhVe6kx2DM9oixAygSJ1uoqwLPcT6w2oF",
   "NodeId":"b9fa5bb4095c0c39e72fe14d9ac3d30e523060ca7a7fbcfddb8afc935da1071f",
   "UserAgent":"",
   "ClientName":"Unknown",
   "ClientOS":"TODO",
   "ClientVersion":"Unknown",
   "Pubkey":"020e488a1189cebf4609fa9558e286b136deed25511e47b513b45158e14c98305e",
   "Addrs":"/ip4/58.96.228.225/tcp/13000/p2p/16Uiu2HAkvPRqnUZx2zKwhVe6kx2DM9oixAygSJ1uoqwLPcT6w2oF",
   "Ip":"58.96.228.225",
   "Country":"Singapore",
   "City":"Singapore",
   "Latency":0,
   "Attempted":false,
   "Succeed":false,
   "Connected":false,
   "Attempts":0,
   "Error":"",
   "MetadataRequest":false,
   "MetadataSucceed":false,
   "ConnectionEvents":null,
   "TotConnections":0,
   "LastConn":0,
   "TotDisconnections":0,
   "LastDisconn":0,
   "TotConnTime":0,
   "ConnFlag":false,
   "LastExport":0,
   "BeaconBlock":{
      "Cnt":0,
      "FirstMessageTime":0,
      "LastMessageTime":0
   },
   "BeaconAggregateProof":{
      "Cnt":0,
      "FirstMessageTime":0,
      "LastMessageTime":0
   },
   "VoluntaryExit":{
      "Cnt":0,
      "FirstMessageTime":0,
      "LastMessageTime":0
   },
   "ProposerSlashing":{
      "Cnt":0,
      "FirstMessageTime":0,
      "LastMessageTime":0
   },
   "AttesterSlashing":{
      "Cnt":0,
      "FirstMessageTime":0,
      "LastMessageTime":0
   }
}
```
I thought too that I had to add the `json: ... ` but it worked fine without them

Reply:
Sorry for arriving late to this discussion. I think that if go writes and reads the json, there is no problem with not adding the `json:____`. The name appearing on the json copies the one of the variable `NameVariable`, which can be modified by adding the `json:___` tag

Reply:
Sounds good thanks, resolving.